tests/AutomationTests/Services/YamlFileServiceTests.cs

```diff
@@ -0,0 +1,106 @@
+﻿// --------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// --------------------------------------------------------------------------------------------
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using Xunit;
+using Microsoft.Oryx.Automation.Models;
+using Microsoft.Oryx.Automation.Services;
+
+namespace Microsoft.Oryx.Automation.Tests.Services
+{
+    public class YamlFileServiceTests
+    {
+        private readonly YamlFileService yamlFileService;
+        private readonly string oryxRootPath;
+        private readonly string testConstantsYamlFilePath;
+
+        public YamlFileServiceTests()
+        {
+            this.oryxRootPath = Directory.GetCurrentDirectory();
+            this.testConstantsYamlFilePath = Path.Combine(this.oryxRootPath, "test-constants.yaml");
+            this.yamlFileService = new YamlFileService(this.oryxRootPath);
```

Comment:
Since we're writing to files in this test suite, we should create a temp directory for the files we're creating so there's no potential crossover of tests targeting the same files or paths. The easiest way to do this would be to generate a random GUID that acts as the intermediate folder between the root path that we're using and any files that we'd be writing to.

Reply:
Updated to use GUID